compute/client_library/snippets/tests/test_ip_address.py

```diff
@@ -268,3 +271,20 @@ def test_release_static_ip(static_ip: Address):
     )
     ips = list_ip_addresses(client, PROJECT, region=region)
     assert static_ip.name not in ips
+
+
+@pytest.mark.parametrize("static_ip", [{"region": "us-central1"}], indirect=True)
+def test_assign_static_external_new_vm(static_ip, disk_fixture):
+    instance_name = "i" + uuid.uuid4().hex[:10]
+    client = AddressesClient()
+    ip_address = client.get(project=PROJECT, region=REGION, address=static_ip.name)
+    instance = assign_static_external_ip_to_new_vm(
+        PROJECT,
+        INSTANCE_ZONE,
+        instance_name,
+        ip_address=ip_address.address,
+    )
+    assert (
+        instance.network_interfaces[0].access_configs[0].nat_i_p == ip_address.address
+    )
+    delete_instance(PROJECT, INSTANCE_ZONE, instance_name)
```

Comment:
Put it in try-finally to make sure deletion happens even if assert raises an error.